server/src/modules/task/task.controller.ts

```diff
@@ -1,4 +1,45 @@
-import { Controller } from '@nestjs/common';
+import {
+  Controller,
+  Get,
+  Post,
+  Body,
+  Delete,
+  Param,
+  Put,
+} from '@nestjs/common';
+import { TaskService } from './task.service';
+import { Task } from './task.model';
+import { TaskDTO as TaskInterface } from './task.interface';
 
 @Controller('task')
-export class TaskController {}
+export class TaskController {
+  constructor(private taskService: TaskService) {}
+
+  @Get('all-task')
+  async findAll(): Promise<TaskInterface[]> {
+    return this.taskService.getAllTask();
+  }
+
+  @Get(':id')
+  async getTaskById(@Param('id') id: number): Promise<Task> {
+    return this.taskService.getTaskById(id);
+  }
+
+  @Post('create')
```

Comment:
Similarly, sometimes people choose `@Post('/')` to represent creating one of the relevant objects in a REST API style if you prefer

Reply:
Did you want to make `taskData` of type `TaskDTO` instead of `Task`? I'm assuming that's what `TaskDTO` was intended for (same in `updateTask` below).